pkg/build/types/definition_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package types
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestNewDefinitionFromURI(t *testing.T) {
```

Comment:
you don't have to change this, but from my point of view, it would be easier to have a table of test cases indicating whether you expect the case to return an error or not.

    cases := []struct{ uri string; shouldFail bool}{{uri: "test", shouldFail: true}, {uri: "://test", shouldFail: false}, ...}

If you do that, you can give each case a name so that it's easier to figure out from the output what's going on.

Reply:
Good point and fixed. It will make the code easier to extend if we decide to add more cases.